### PR TITLE
qrq: init at 0.3.5

### PR DESCRIPTION
**Pull request by pkharvey**
## Description of changes

The reason for the patches is because the source code contains hard-coded magic numbers for path lengths, which crop nix store paths.

## Things done

- Built on platform(s)
  - [X] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/qr/qrq/package.nix [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+  makeFlags = [
+    "BUILD_INFO=nix"
+    "DESTDIR=${builtins.placeholder "out"}"
+  ];
+
+  prePatch = ''
+    cd src
+    substituteInPlace qrq.c --replace-fail '[80]' '[4000]'
+    substituteInPlace qrq.c --replace-fail '80,' '4000,'
+  '';
```
Comment (sikmir) [inline pkgs/by-name/qr/qrq/package.nix:34]:
```suggestion
  postPatch = ''
    substituteInPlace src/qrq.c \
      --replace-fail '[80]' '[4000]' \
      --replace-fail '80,' '4000,'
  '';
```

File: pkgs/by-name/qr/qrq/package.nix [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+  meta = {
+    description = "High Speed Morse trainer - mirror of https://git.fkurz.net/dj1yfk/qrq";
+    homepage = "https://github.com/dj1yfk/qrq";
+    changelog = "https://github.com/dj1yfk/qrq/blob/${src.rev}/ChangeLog";
+    license = lib.licenses.gpl2;
```
Comment (sikmir) [inline pkgs/by-name/qr/qrq/package.nix:40]:
```suggestion
    license = lib.licenses.gpl2Plus;
```

**Comment by pkharvey** [comment]
`qrq` is working fine, however I have tried to get `qrqscore` to run with perl using the following diff

```
diff --git a/pkgs/by-name/qr/qrq/package.nix b/pkgs/by-name/qr/qrq/package.nix
index 47113716ab8e..83ebcbe8cf2a 100644
--- a/pkgs/by-name/qr/qrq/package.nix
+++ b/pkgs/by-name/qr/qrq/package.nix
@@ -3,6 +3,7 @@
   stdenv,
   fetchFromGitHub,
   ncurses,
+  perl,
   pulseaudio,
 }:
 
@@ -19,6 +20,7 @@ stdenv.mkDerivation rec {
 
   buildInputs = [
     ncurses
+    (perl.withPackages (ps: with ps; [ LWP ]))
     pulseaudio
   ];
```

but the perl script seems to get stuck in a loop and I can't yet work out why.  I'm still working on this.

Also, I learned `gnuplot` enables a feature of the program to graph the scores, so I could include this also.

Best to merge this now and fix in the future.

**Comment by MatthewCroughan** [comment]
Yeah, this is an obscure ham-radio package that we're interested in maintaining, best to just merge it and add the optional features in later.

**Review by MatthewCroughan — APPROVED** [review]

**Comment by ghpzin** [comment]
Build on Hydra failed, looking at ofborg results it failed the same way:
https://hydra.nixos.org/job/nixos/trunk-combined/nixpkgs.qrq.x86_64-linux
https://hydra.nixos.org/build/274337023
```text
Running phase: unpackPhase
unpacking source archive /nix/store/wvpy8nmhz7f840qww6kmlbbsinr58nir-source
source root is source
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
build flags: SHELL=/nix/store/izpf49b74i15pcr9708s3xdwyqs4jxwl-bash-5.2p32/bin/bash BUILD_INFO=nix DESTDIR=/nix/store/0f5bmv1vh1a426v563vhjwwy1p1sq7ym-qrq-0.3.5
make: *** No targets specified and no makefile found.  Stop.
```
It probably needs something to specify that `Makefile` is in `src` subdir and build from it.
This makes it build, thought there might be a better way I am not aware of:
```diff
diff --git a/pkgs/by-name/qr/qrq/package.nix b/pkgs/by-name/qr/qrq/package.nix
index 47113716ab..e4b50309ec 100644
--- a/pkgs/by-name/qr/qrq/package.nix
+++ b/pkgs/by-name/qr/qrq/package.nix
@@ -22,6 +22,8 @@
     pulseaudio
   ];

+  preBuild = "cd src";
+
   makeFlags = [
     "BUILD_INFO=nix"
     "DESTDIR=${builtins.placeholder "out"}"
```

**Comment by MatthewCroughan** [comment]
@ghpzin thanks a lot for figuring that out, I'm not sure how I didn't catch this earlier. Amended in https://github.com/NixOS/nixpkgs/pull/346888, the better way is to use the `sourceRoot` variable.

**Pull request merged**